libs/fairness/src/lib/Controls/OutcomePlot.tsx

```diff
@@ -128,22 +165,77 @@ export class OutcomePlot extends React.PureComponent<IOutcomePlotProps> {
           )
         } as any
       ];
-      outcomeChartModalHelpStrings = [
+      outcomeChartCalloutHelpBarStrings = [
         localization.Fairness.Report.regressionOutcomesHowToRead
       ];
     }
 
+    barPlotlyProps.data[0].customdata = [];
+    const digitsOfPrecision = 1;
+
+    for (let i = 0; i < this.props.dashboardContext.groupNames.length; i++) {
+      const tempY = barPlotlyProps.data[0].y
+        ? barPlotlyProps.data[0].y[i]
+        : undefined;
+      const tempX = barPlotlyProps.data[0].x
+        ? barPlotlyProps.data[0].x[i]
+        : undefined;
+      const x =
+        tempX !== undefined && typeof tempX == "number" ? tempX : undefined;
```

Comment:
What's going on here? It should not be a number? Can you add a comment to explain the reasoning?

Reply:
It should be a number. Ke recommended to do type checking instead of type casting